### PR TITLE
Fix simulation id validation

### LINKED ISSUE #127: Can't run the PI with only coupled/group PI parameters
When trying to run a parameter identification I ran into a error: 
"Error in .validateSimulationIds(ids, parameters, outputMappings) :  Mismatch or missing ID detected: [...]. Ensure each Simulation ID matches with corresponding PIParameter and OutputMapping IDs."

It seems to be happening when only grouped PI parameters are given.

How to reproduce:
Using the example in the [user-guide](https://www.open-systems-pharmacology.org/OSPSuite.ParameterIdentification/articles/user-guide.html), trying to run only on the lipophilicity doesn't work:

works (example in the user-guide): 
```
piTask <- ParameterIdentification$new(
  simulations = list(sim_250mg, sim_500mg),
  parameters = list(piParameterLipo, piParameterCl_250mg, piParameterCl_500mg),
  outputMappings = list(outputMapping_250mg, outputMapping_500mg),
  configuration = piConfiguration
)
``` 

doesn' t work: 
```
piTask <- ParameterIdentification$new(
  simulations = list(sim_250mg, sim_500mg),
  parameters = piParameterLipo,
  outputMappings = list(outputMapping_250mg, outputMapping_500mg),
  configuration = piConfiguration
)
```
@rengelke

### PR DESCRIPTION
**Pull request by rengelke**
Fixes #127 where validation did not consider that piParams can have multiple levels

### CODE REVIEW

File: R/utilities-simulation.R [hunk 1]
```diff
@@ -31,19 +31,30 @@
...
-      stop(messages$errorSimulationIdMissing(id))
-    }
+  # sort IDs before comparison
+  simulationIds <- sort(unique(unlist(simulationIds)))
+  piParamIds <- sort(piParamIds)
+  outputMappingIds <- sort(outputMappingIds)
+
+  # Validate that simulationId is idnetical with piParamIds and outputMappingIds
+  if (!identical(simulationIds, piParamIds) || !identical(simulationIds, outputMappingIds)) {
+    stop(messages$errorSimulationIdMissing(id))
```
Comment (dlefaudeux) [inline R/utilities-simulation.R:57]: The `id` is not defined here anymore

File: R/utilities-simulation.R [hunk 2]
```diff
@@ -31,19 +31,30 @@
...
-  for (id in simulationIds) {
-    if (!(id %in% piParamIds && id %in% outputMappingIds)) {
-      stop(messages$errorSimulationIdMissing(id))
-    }
+  # sort IDs before comparison
+  simulationIds <- sort(unique(unlist(simulationIds)))
+  piParamIds <- sort(piParamIds)
+  outputMappingIds <- sort(outputMappingIds)
+
+  # Validate that simulationId is idnetical with piParamIds and outputMappingIds
```
Comment (dlefaudeux) [inline R/utilities-simulation.R:55]:
```suggestion
  # Validate that simulationId is identical with piParamIds and outputMappingIds
```

**Review by dlefaudeux — COMMENTED** [review]
I think you should also add more tests, for piParameters and OutputMappings with sim ids not in the given simulationIds

**Comment by dlefaudeux** [comment]
Also fixes #137

**Review by dlefaudeux — APPROVED** [review]
Added test should be good enough.

**Review by Felixmil — APPROVED** [review]

File: R/utilities-simulation.R [hunk 3]
```diff
@@ -31,19 +31,34 @@
...
-    if (!(id %in% piParamIds && id %in% outputMappingIds)) {
-      stop(messages$errorSimulationIdMissing(id))
-    }
+  # sort IDs before comparison
+  simulationIds <- sort(unique(unlist(simulationIds)))
+  piParamIds <- sort(piParamIds)
+  outputMappingIds <- sort(outputMappingIds)
+
+  # Validate that simulationId is identical with piParamIds and outputMappingIds
+  if (!identical(simulationIds, piParamIds) || !identical(simulationIds, outputMappingIds)) {
```
Comment (PavelBal) [inline R/utilities-simulation.R:58]: When could this happen? How could the user invoke this behavior? @rengelke

Comment (dlefaudeux) [reply R/utilities-simulation.R:58]: If the user forget data or parameters or simulations in the ParameterIdentification$new(), for example.
cf added tests

**Pull request merged**